config/v1/types_console.go

```diff
@@ -26,6 +29,26 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// customization is used to optionally provide a small set of
+	// customization options to the web console.
+	// +optional
+	Customization ConsoleCustomization `json:"customization,omitempty"`
+}
+
+// ConsoleCustomization defines a list of optional configuration for the console.
+type ConsoleCustomization struct {
+	// customURL contains URL and certificate for replacing the default console URL.
+	// +optional
+	CustomURL `json:customURL,omitempty`
+}
+
+// CustomURL replaces the default console URL under which console will be available.
+type CustomURL struct {
+	// host is the desired custom URL under which console will be available.
+	Host string `json:host:omitempty`
+	// servingCert references to Secret in the openshift-config namespace that
+	// contains custom certificate and key.
```

Comment:
We need to document the fields in the referenced secret that contain the certificate and key.

Reply:
@spadgett you mean the fields on the Secret itself - `tls.crt`and `tls.key`
Or the `Name` and the `Namespace` and where they should point to ?

Reply:
I mean the keys in the secret, `tls.crt` and `tls.key`. That needs to be documented somewhere, otherwise the user has no way to know.

Reply:
Yeah as the Operator will have to pluck them out by this.  Unless we let the user configure the keys, but that seems unnecessary.